core/src/test/java/io/smallrye/openapi/runtime/io/JacksonJsonIOTest.java

```diff
@@ -0,0 +1,19 @@
+package io.smallrye.openapi.runtime.io;
+
+import org.junit.jupiter.api.BeforeEach;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+class JacksonJsonIOTest extends JsonIOTest<JsonNode, ArrayNode, ObjectNode, ArrayNode, ObjectNode> {
+
+    @BeforeEach
+    void setup() {
+        super.target = new JacksonJsonIO(new ObjectMapper()
+                .enable(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS));
```

Comment:
Is `USE_BIG_DECIMAL_FOR_FLOATS` just needed to get predictable accuracy for the tests? We don't appear to use it anywhere else in the code.

Reply:
Hmm, I assume this will change the value that `JsonIO.fromJson(JsonNode)` returns for a decimal number from a `double` to a `BigDecimal`. I guess usually we'll call `JsonIO.fromJson(JsonNode, DataType)` instead though.

It seems odd to be testing with an option that we don't enable at runtime, though the user or integrator could provide their own `ObjectMapper` and I suppose we're trying to test the path where `fromJson(JsonNode)` does return a `BigDecimal`.

Reply:
> It seems odd to be testing with an option that we don't enable at runtime, though the user or integrator could provide their own `ObjectMapper` and I suppose we're trying to test the path where `fromJson(JsonNode)` does return a `BigDecimal`.

Right - it's contrived for the test. As you said we call the overload with the expected type usually, and this is the fallback. It would also be used for things like examples or extensions. It's not really possible for an integrator to provide the `ObjectMapper` cleanly yet, but that is something that could be available at some point. It would be useful in cases like the test here where a `double` would truncate the user's value in an example/extension.